File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/bit-manipulation.java
/**
 * Java Implementation: Bit Manipulation
 * Invariant: Bitwise AND, OR, XOR, Shifts O(1)
 */
package seed.dsa.core;

import java.util.*;

public class BitManipulation<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/cpp/bit-manipulation.cpp
/**
 * C++ Implementation: Bit Manipulation
 * Invariant: Bitwise AND, OR, XOR, Shifts O(1)
 */
#include <iostream>
#include <vector>
#include <stdexcept>

template <typename T>
class BitManipulation {
private:
    std::vector<T> elements;
public:
    // TODO: Implement primary invariant operations
    void push(const T& val) {
        elements.push_back(val);
    }
    size_t size() const { return elements.size(); }
};
